ocaml/libs/xapi-stdext/lib/xapi-stdext-unix/unixext.ml

```diff
@@ -547,7 +547,8 @@ let really_read_string fd length =
 
 exception Timeout
 
-let to_milliseconds ms = ms *. 1000. |> ceil |> int_of_float
+let to_milliseconds s =
+  Mtime.Span.to_float_ns s *. 1e-6 |> Float.ceil |> int_of_float
```

Comment:
This is not the same functionality? The old code was confusing using `ms` as argument when it probably was seconds. What is the unit of the argument now? seconds, I assume. Why do we need `Mtime.Span.to_float_ns`?

Reply:
Unfortunately there is no way to directly convert an `Mtime.Span.t` to seconds. The only conversion we can do is to nanoseconds, which we then need to convert into the units that we desire (seconds, or milliseconds, in this case milliseconds, because that is what Polly expects).

Reply:
I don't think there's a need to use floats here (spans are always positive)
```suggestion
Int64.div (Mtime.Span.to_uint_ns s) 1_000_000L |> Int64.to_int
```

Reply:
Using int64 will be a little bit slower (it needs to allocate a boxed int64, both `int` and `float` are unboxed), but in this context it probably doesn't matter, and might be more accurate.

Reply:
Changed to use int64.

Reply:
Actually this doesn't work, because it doesn't round correctly, I'll fix it.

Reply:
This works, and I added a comment explaining why (I had the very same bug in the original float version, which is why I introduced `Float.ceil`):
```
+(* Need to round up when converting to milliseconds:
+   The span will have the amount of time already elapsed subtracted, so if we are asked to wait
+   for 1ms, the span will be slightly smaller than that.
+   If we truncate then all waits of 1ms will be 0ms instead, which would be wrong, as we wouldn't wait at all.
+   If we round to nearest rather than up, then waits of 1ms will be ~0.5ms instead, since values below 0.5ms will get
+   rounded to 0.
+   Rounding up works correctly, and makes the tests pass.
+ *)
+let to_milliseconds span =
+  Int64.div (Int64.add 999_999L @@ Mtime.Span.to_uint64_ns span) 1_000_000L |> Int64.to_int
```